game/atmosphere.rpy

```diff
@@ -49,168 +135,458 @@ init 0 python in jn_atmosphere:
+        night_clouds_image="clouds night heavy",
+        day_particles_image="particles rain",
+        night_particles_image="particles rain",
         weather_sfx="mod_assets/sfx/rain_muffled.mp3"
     )
 
     WEATHER_THUNDER = JNWeather(
         weather_type=JNWeatherTypes.thunder,
-        day_image="sky_day thunder",
+        day_sky_image="sky day thunder",
+        night_sky_image="sky night thunder",
+        notify_text=[
+            "Those clouds look really dark, huh? :<",
+            "It was a dark and stormy night...",
+            "Ugh... I hate storms... {0}".format(jn_utils.getRandomAngryEmoticon()),
+            "Hey... is it stormy over there too? :/",
+        ],
         dim_image="dim heavy",
+        day_clouds_image="clouds day thunder",
+        night_clouds_image="clouds night heavy",
+        day_particles_image="particles rain",
+        night_particles_image="particles rain",
         weather_sfx="mod_assets/sfx/rain_muffled.mp3"
     )
 
+    WEATHER_SNOW = JNWeather(
+        weather_type=JNWeatherTypes.snow,
+        day_sky_image="sky day snow",
+        night_sky_image="sky night overcast",
+        notify_text=[
+            "[player]! [player]! It's snowing! {0}".format(jn_utils.getRandomHappyEmoticon()),
+            "It's snowing! It's snowing! {0}".format(jn_utils.getRandomHappyEmoticon()),
+            "[player]! Is it snowing for you too?! {0}".format(jn_utils.getRandomHappyEmoticon()),
+        ],
+        dim_image="dim light",
+        day_clouds_image="clouds day light",
+        night_clouds_image="clouds night light",
+        day_particles_image="particles snow",
+        night_particles_image="particles snow",
+    )
+
     WEATHER_SUNNY = JNWeather(
         weather_type=JNWeatherTypes.sunny,
-        day_image="sky_day sunny"
+        day_sky_image="sky day sunny",
+        night_sky_image="sky night sunny",
+        day_clouds_image="clouds day light",
+        night_clouds_image="clouds night light"
     )
 
     WEATHER_GLITCH = JNWeather(
         weather_type=JNWeatherTypes.glitch,
-        day_image="glitch_fuzzy")
-
-    __RANDOM_WEATHER_TYPES = [
-        WEATHER_OVERCAST,
-        WEATHER_RAIN,
-        WEATHER_THUNDER,
-        WEATHER_SUNNY
-    ]
+        day_sky_image="glitch_fuzzy",
+        night_sky_image="glitch_fuzzy")
+
+    # Weather code -> JNWeather map
+    # key: Regex matching the weather code as a string, allowing ranged captures (returned from OpenWeatherMap)
+    # value: JNWeather associated with the weather code range
+    __WEATHER_CODE_REGEX_TYPE_MAP = {
+        ("^2[0-9][0-9]$"): WEATHER_THUNDER, # Thunder
+        ("^3[0-9][0-9]$"): WEATHER_RAIN, # Drizzle
+        ("^5[0-9][0-9]$"): WEATHER_RAIN, # Rain
+        ("^6[0-9][0-9]$"): WEATHER_SNOW, # Snow
+        ("^7[0-9][0-9]$"): WEATHER_OVERCAST, # Misc (mist, tornado, sandstorms, etc.)
+        ("(800|801|802|803)"): WEATHER_SUNNY, # Clear/light clouds
+        ("(804)"): WEATHER_OVERCAST, # Clouds
+    }
```

Comment:
Just looks inflexible to me. Like what if we would be able to control rain amount one day.

Also, can't we just figure out rain intensity (idk how it's called) directly `lower < intensity < upper`?

And if you're sure this is the approach you want, then question, why do we do `"^5[0-9][0-9]$"` instead of `"^5\d{2}$"`? Although, I'd really think about just comparing the result code.

Reply:
I'd personally be happy to settle for the regex checks; I totally get where you're coming from but I don't see weather becoming that granular: at most, I'd expect we might introduce a weaker rain type, but then we could add WEATHER_LIGHT_RAIN and just replace the drizzle map value.

Good catch with the regex though! I can definitely get those replaced. Blame that on my regex skills being trash